src/main/java/com/troubleshooting/troubleshootingtool/ContainerDataRetrieval.java

```diff
@@ -0,0 +1,101 @@
+package com.troubleshooting.troubleshootingtool;
+
+import com.troubleshooting.model.ConjurLogModel;
+import com.troubleshooting.model.EnvironmentModel;
+import com.troubleshooting.model.EnvironmentCollectionModel;
+import com.troubleshooting.model.LogCollectionModel;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+/**
+ * Retrieves the data returned from the Docker Client and packs the data in the appropriate data models
+ */
+@Component
+public final class ContainerDataRetrieval implements DataRetrieval {
+    private ContainerDataAccess access;
+    private LogCollectionModel logCollection;
+    private EnvironmentCollectionModel environmentCollection;
+
+    static String STDERR = "STDERR";
+
+    @Autowired
+    public ContainerDataRetrieval(ContainerDataAccess access, LogCollectionModel logCollection, EnvironmentCollectionModel environmentCollection) {
+        this.access = access;
+        this.logCollection = logCollection;
+        this.environmentCollection = environmentCollection;
+    }
+
+    @Override
+    public LogCollectionModel getLogs(String containerID, String query) {
+        List<String> loggingFrames = new ArrayList<String>();
```

Comment:
Depending on your java level you can avoid `new ArrayList<String>();'` and have `new ArrayList<>();`